Add version to commands

miketheman opened the pull request:
Add a helper to show the current git SHA of the installed git repo.

Signed-off-by: Mike Fiedler <miketheman@gmail.com>

miketheman commented:
This is a first pass at a helper function to show details about the installed code, as related and pertains to the discussion in #1038

The current output of this command shows:

```console
$ bash-it version
Current git SHA:
3f839d5 on 2017-09-29T08:19:19+02:00
https://github.com/Bash-it/bash-it/commit/3f839d5ac28305cc83eaa3555973900e724d5a34
```

miketheman commented on lib/helpers.bash:
```diff
@@ -90,8 +91,10 @@ bash-it ()
              return;;
          update)
              func=_bash-it_update;;
-          migrate)
```
A little OCD alignment here.

nwinkler replied:
Thanks for catching that one!

tswicegood replied:
🎉

miketheman commented:
Not at all! I'll try to figure that out over then next 48 hours.

miketheman commented:
@tswicegood Done! Let me know if you'd prefer me to squash these, or if you'd like to do that with the green button here.

nwinkler commented on lib/helpers.bash:
```diff
@@ -227,6 +230,21 @@ _bash-it-migrate() {
+  echo "Current git SHA:"
+  echo "$(git log --pretty=format:'%h on %aI' -n 1)"
+
+  CURRENT_GIT_SHA=$(git rev-parse --verify HEAD)
```
It might be quicker to take the result of the previous `git log` command and then just take the first part of that (e.g. with a Bash Regex or `sed`), instead of forking another `git` command.

I haven't measured any timing, but I would assume that getting the first part of the variable would be quicker than running another `git` command...

miketheman replied:
True - I was using them differently - one was a shortlog, the other a long commit SHA. Since github links  and git log work with a short hash just as well, I can make that change.

nwinkler commented on lib/helpers.bash:
```diff
@@ -227,6 +230,21 @@ _bash-it-migrate() {
+  echo "$(git log --pretty=format:'%h on %aI' -n 1)"
+
+  CURRENT_GIT_SHA=$(git rev-parse --verify HEAD)
+  echo "https://github.com/Bash-it/bash-it/commit/$CURRENT_GIT_SHA"
```
What about forks in this case? Does it make sense to always refer to the `Bash-it/bash-it` location?

miketheman replied:
Likely not. I can add some logic to handle this, it incurs another subshell to `git` - but from some naive timings, the entire command lasts less than 200ms, since I'm replacing the other subshell.

miketheman commented:
@nwinkler I've addressed the comments, and updated some variable names for namespace.

We still end up with two subshell calls - I ran it through a sample [timing command](http://www.bashoneliners.com/oneliners/oneliner/48/):

```bash
$ for i in {1..100}; do time bash-it version; done 2>&1 | grep ^real | sed -e s/.*m// | awk '{sum += $1} END {print sum / NR}'
0.07485
```

So it's likely to still be pretty fast.

nwinkler commented:
Looks good to me now, thanks! @tswicegood Ready to merge?

tswicegood commented:
Thanks!!
